### PR TITLE
complete GET/api/articles/:author_id and reworked GET/api tests

### PR DESCRIPTION
**Pull request by Pirkhs**
Completed the GET endpoint for /api/articles/:author_id and also reworked the tests for GET/api using the test data from the actual endpoints.json file

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -24,30 +25,61 @@ describe('GET /api/topics', () => {
...
+
+    test('STATUS 404: responds with appropriate error message for when given a valid but non-existant id in the data', () => {
+        return request(app)
+        .get("/api/articles/99999")
+        .expect(404)
+        .then(result => {
+            expect(result.body.msg).toBe("Not found")
+        })
+    })
+});
```
Comment (nc-mezz-davies) [inline __tests__/app.test.js:78]: Nice!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -24,30 +25,61 @@ describe('GET /api/topics', () => {
...
     });
 });
 
+describe('GET /api/articles:article_id', () => {
+    test('STATUS 200: responds with an article object, which should have its appropriate properties ', () => {
+        return request(app)
+        .get("/api/articles/1")
+        .expect(200)
+        .then(result => {
+            const {article} = result.body
```
Comment (nc-mezz-davies) [inline __tests__/app.test.js:49]: The article should not be returned in an array back to the user. Arrays are for more than one item or potentially more than one item. We know that there will only every be one article returned so we should just return that. Please achieve this by rewriting your test then updating your code to pass it. TDD remember!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -24,30 +25,61 @@ describe('GET /api/topics', () => {
...
+            expect(article[0].article_id).toBe(1)
```
Comment (nc-poppy-m) [inline __tests__/app.test.js:50]: This should be dexonstructed in the model

Comment (nc-poppy-m) [inline __tests__/app.test.js:50]: Thorough tests for this 200, good job!

File: __tests__/app.test.js [hunk 4]
```diff
@@ -24,30 +25,61 @@ describe('GET /api/topics', () => {
...
+            expect(article[0].topic).toBe("mitch")
+            expect(article[0].author).toBe("butter_bridge")
+            expect(article[0].body).toBe('I find this existence challenging')
+            expect(article[0].created_at).toBe("2020-07-09T20:11:00.000Z")
+            expect(article[0].votes).toBe(100)
+            expect(article[0].article_img_url).toBe("https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700")
+        })   
+    });
+
+    test('STATUS 400: responds with appropriate error message for when given an invalid id', () => {
```
Comment (nc-poppy-m) [inline __tests__/app.test.js:61]: Excellent!

File: models/app.models.js [hunk 5]
```diff
@@ -14,4 +14,12 @@ exports.readAllEndpoints = () => {
     return fs.readFile("./endpoints.json", "utf-8").then(endpointsData => {
         return endpointsData
     })
+}
+
+exports.selectArticleById = (article_id) => {
+    return db.query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
+    .then(article => {
+        if (article.rows.length === 0) return Promise.reject({status: 404, msg: "Not found"})
+        return article.rows
```
Comment (nc-poppy-m) [inline models/app.models.js:23]: If selecting hte first one, ensure you get `rows[0]` to only return what you need

**Review by nc-poppy-m — COMMENTED** [review]
Good work!

**Review by nc-mezz-davies — COMMENTED** [review]
Your endpoints functionality is fine 👍 . Please have a look at my comment on GET /api/articles/:article_id

**Pull request merged**